src/app/staticanalysis/modules/checksEffectsInteraction.js

```diff
@@ -0,0 +1,83 @@
+var name = 'Checks-Effects-Interaction pattern'
+var desc = 'Avoid potential reentrancy bugs'
+var categories = require('./categories')
+var common = require('./staticAnalysisCommon')
+var callGraph = require('./functionCallGraph')
+var AbstractAst = require('./abstractAstView')
+
+function checksEffectsInteraction () {
+  this.contracts = []
+
+  checksEffectsInteraction.prototype.visit = new AbstractAst().builder(
+    (node) => common.isInteraction(node) || common.isEffect(node) || common.isLocalCallGraphRelevantNode(node),
+    this.contracts
```

Comment:
how would you return a list, from which function? visit has no return value it is ignored by the static analysis runner.